relentless/simulate/lammps.py

```diff
@@ -0,0 +1,347 @@
+        super().__init__(neighbor_buffer, units, atom_style)
+        self.filename = filename
+
+    def to_commands(self, sim):
+        cmds = super().to_commands(sim)
+        cmds += ['read_data {filename}'.format(filename=self.filename)]
+        cmds += self.attach_potentials(sim)
+
+        return cmds
+
+class InitializeRandomly(Initialize):
+    def __init__(self, seed, neighbor_buffer, units='lj', atom_style='atomic'):
+        super().__init__(neighbor_buffer, units, atom_style)
+        self.seed = seed
+
+    def to_commands(self, sim):
+        cmds = super().to_commands(sim)
+
+        # make box from ensemble
+        box = self.extract_box_params(sim)
+        if not np.all(np.isclose(box[-3:],0)):
+            cmds += ['region box prism {} {} {} {} {} {} {} {} {}'.format(*box)]
+        else:
+            cmds += ['region box block {} {} {} {} {} {}'.format(*box[:-3])]
+        cmds += ['create_box {N} box'.format(N=len(sim.ensemble.types))]
+
+        # use lammps random initialization routines
+        for i in sim.ensemble.types:
+            if sim.ensemble.N[i] is None:
+                raise ValueError('Number of particles for type {} must be set.'.format(i))
+
+            cmds += ['create_atoms {typeid} random {N} {seed} box'.format(typeid=sim.type_map[i],
+                                                                          N=sim.ensemble.N[i],
+                                                                          seed=self.seed+sim.type_map[i]-1)]
+        cmds += ['mass * 1.0',
+                 'velocity all create {temp} {seed}'.format(temp=sim.ensemble.T,
+                                                            seed=self.seed)]
+
+        cmds += self.attach_potentials(sim)
+
+        return cmds
+
+class MinimizeEnergy(LAMMPSOperation):
+    def __init__(self, energy_tolerance, force_tolerance, max_iterations, dt):
+        self.energy_tolerance = energy_tolerance
+        self.force_tolerance = force_tolerance
+        self.max_iterations = max_iterations
+
+    def to_commands(self, sim):
+        cmds = ['minimize {etol} {ftol} {maxiter} {maxeval}'.format(etol=self.energy_tolerance,
+                                                                    ftol=self.force_tolerance,
+                                                                    maxiter=self.max_iterations,
+                                                                    maxeval=100*self.max_iterations)]
+
+        return cmds
+
+class AddLangevinIntegrator(LAMMPSOperation):
+    def __init__(self, dt, friction, seed):
+        self.friction = friction
+        self.seed = seed
+
+        self._fix_langevin = self.new_fix_id()
+        self._fix_nve = self.new_fix_id()
+
+    def to_commands(self, sim):
+        # obtain per-type mass
+        Ntypes = len(sim.ensemble.types)
+        _mass = sim.lammps.numpy.extract_atom('mass')
+        if _mass is None or _mass.shape != (Ntypes+1,1):
```

Comment:
Is `_mass` really being returned as a 2d array? If so, we should coerce it into a 1d array so that its shape is `(Ntypes+1,)`. Maybe try specifying `dim=1` in `extract_atom`?

Reply:
There's also `np.squeeze` if you want to just smoosh the array down.

https://numpy.org/doc/stable/reference/generated/numpy.squeeze.html